### ISSUE TITLE
rpm macros cannot process long options

### ISSUE THREAD

**Issue by rhabacker** [rpm-software-management/rpm#2541]
With actual rpm releases (Leap: 4.14, Tumbleweed 4:18) it is not possible to parse options in the form `--<key>` or `--<key> arg` as shown below:

```
$ rpm --version
RPM version 4.14.3
$ LC_ALL=C rpm --define '%foo(-) "%**"' --eval '%foo --test file1'
foo: invalid option -- '-'
error: Unknown option - in foo(_:)
```
Trying to parse `--<key> arg` returns:
```
LC_ALL=C rpm --define '%foo(-:) "%**"' --eval '%foo --test arg  file1'
foo: invalid option -- '-'
error: Unknown option - in foo(_:)
```

With rpm 4.18 the result is:
```
$ LC_ALL=C ../rpm-build/rpm 
RPM version 4.18.90
$ LC_ALL=C ../rpm-build/rpm --define '%foo(-) "%**"' --eval '%foo --test  file1'
"--test file1"
```
and 
```
LC_ALL=C ../rpm-build/rpm --define '%foo(-:) "%**"' --eval '%foo --test arg  file1'
error: Unknown option - in foo(-:)
```

**Comment by pmatilai** [comment]
I think it's quite well known, and at least implicitly documented via the parametric macro declaration syntax, that only short option names are supported by rpm itself. Passing `-` as the option string delegates *all* option processing to the macro (as per https://rpm-software-management.github.io/rpm/manual/macros.html), and from there on you can do pretty much anything you want (at least from Lua).  `-:` is simply invalid.

One could of course think of this as an RFE to add support for long options, but I don't see that happening, so there's no point keeping this open.

**Comment by rhabacker** [comment]
> Passing `-` as the option string delegates _all_ option processing to the macro (as per https://rpm-software-management.github.io/rpm/manual/macros.html),

This is not true for rpm version 4.14 (which is the default on Leap) as mentioned in the first report, it simply outputs 

> $ rpm --define '%foo(-) "%**"' --eval '%foo --test'
> foo: invalid option -- '-'
> error: Unknown option - in foo(_:)

**Comment by rhabacker** [comment]
> ... at least implicitly documented via the parametric macro declaration syntax, that only short option names are supported by rpm itself. 

This could be formulated even more clearly. 

> Passing `-` as the option string delegates _all_ option processing to the macro (as per https://rpm-software-management.github.io/rpm/manual/macros.html), 

This seems to have been added with #1392, and since it's a workaround for not supporting long options, I think it's worth adding a note that it only works with rpm >= 4.17 and how the shell-like macros are set up.

Looking at the code, it looks like %# is set to the number of arguments, each single argument is put into %1..%n and all arguments are merged into %* and %**; the macros %{-x..} macros are not defined.

> %1, %2, … 	the arguments themselves (after getopt(3) processing)

This statement is not true for this case.

**Comment by pmatilai** [comment]
> But is not true for rpm version 4.14

4.14 fell out of upstream "support" something like three years ago, it's not really relevant at all. Distros have their own realities of course, that's just how it goes.

Not that it hurt anything to add the version to `-` support in the docs, we usually do that for new features and it was just forgotten in this case. Thanks for that patch.
